pallets/clusters/src/lib.rs

```diff
@@ -29,40 +29,50 @@ pub struct ClusterMember {
 /// The **settings** of a **Role**
 #[derive(Encode, Decode, Clone, PartialEq, Debug, Eq, scale_info::TypeInfo)]
 pub struct RoleSettings {
+	/// Name of the setting
 	pub name: Vec<u8>,
+	/// The data associated to the Role
 	pub data: Vec<u8>,
 }
 
 /// **Struct** of **Role** belonging to a **Cluster**.
 #[derive(Encode, Decode, Clone, PartialEq, Debug, Eq, scale_info::TypeInfo)]
 pub struct Role {
+	/// Name of the role
 	pub name: Vec<u8>,
+	/// Settings of the Role
 	pub settings: RoleSettings,
```

Comment:
can't we just use a map?

Reply:
I removed `RoleSetting` from the `Role` structure and added a storage [`RoleToSettings`](https://github.com/fragcolor-xyz/clamor/blob/8276f14d8e8d6eb42a1fcc163adbc348619888ed/pallets/clusters/src/lib.rs#L112).